src/test/java/com/exasol/adapter/dialects/snowflake/SnowflakeSqlDialectTest.java

```diff
@@ -90,40 +76,41 @@ void testSupportsJdbcSchemas() {
         assertThat(this.dialect.supportsJdbcSchemas(), equalTo(StructureElementSupport.MULTIPLE));
     }
 
-    @Test
+    @Test // TODO: if 'use schemaname' is executed in the current session then it doesnt need it. question: does it
+          // store the session?
     void testRequiresCatalogQualifiedTableNames() {
         assertThat(this.dialect.requiresCatalogQualifiedTableNames(null), equalTo(false));
     }
 
-    @Test
+    @Test // TODO: if 'use dbname' is executed in the current session then it doesnt need it. question: does it store
+          // the session?
     void testRequiresSchemaQualifiedTableNames() {
-        assertThat(this.dialect.requiresSchemaQualifiedTableNames(null), equalTo(true));
+        assertThat(this.dialect.requiresSchemaQualifiedTableNames(null), equalTo(false));
     }
 
-    @Test
+    @Test // NULLS_SORTED_AT_START is default for Snowflake unless specified otherwise
```

Comment:
Is there a way to auto-detect this? Check database metadata!

Reply:
I am not aware of any way to auto-detect it, but based on the [doc](https://docs.snowflake.com/en/sql-reference/constructs/order-by.html#usage-notes) it comes at the top if no sort order is specified.